### PR TITLE
remove ts3.2 versions; TS 3.2 is not supported on DT

### PR DESCRIPTION
**Pull request by sandersn**
TS 3.2 is not supported on DT anymore, so remove the two ts3.2/ subdirectories with explicit support for it.

### CODE REVIEW

**Review by ExE-Boss — APPROVED** [review]

File: types/function-bind/ts3.2/implementation.d.ts [hunk 1]
```diff
@@ -1,183 +0,0 @@
-//#region bind():
-/**
- * For a given function, creates a bound function that has the same body as the original function.
- * The this object of the bound function is associated with the specified object, and has the specified initial parameters.
- * @param thisArg The object to be used as the this object.
- * @param args Arguments to bind to the parameters of the function.
- */
-declare function bind<T, A extends any[], R>(this: (this: T, ...args: A) => R, thisArg: T): (...args: A) => R;
```
Comment (ExE-Boss) [inline types/function-bind/ts3.2/implementation.d.ts:8]: **TS 3.3+** supports `ThisParameterType` and `OmitThisParameter`:
<https://github.com/DefinitelyTyped/DefinitelyTyped/blob/b21787307a30f1d7b7d8e80bc58517e3576266bf/types/function-bind/ts3.9/implementation.d.ts#L8>

File: types/function-bind/ts3.2/function-bind-tests.ts [hunk 2]
```diff
@@ -1,88 +0,0 @@
...
-    constructor(public string: string, public number: number) {}
-    static bind: typeof bind;
-}
-
-// bind.call():
-// $ExpectType new (string: string, number: number) => Foo
-bind.call(Foo, null);
-
-// $ExpectType new (string: string, number: number) => Foo
-Foo.bind(null);
```
Comment (ExE-Boss) [inline types/function-bind/ts3.2/function-bind-tests.ts:43]: This preserves the class type in **TS 3.3+**:
<https://github.com/DefinitelyTyped/DefinitelyTyped/blob/b21787307a30f1d7b7d8e80bc58517e3576266bf/types/function-bind/ts3.9/function-bind-tests.ts#L42-L43>

File: types/function-bind/ts3.2/function-bind-tests.ts [hunk 3]
```diff
@@ -1,88 +0,0 @@
...
-// $ExpectType (end?: number | undefined) => any[]
-expectType<(end?: number) => any[]>(bind.call(Function.call, Array.prototype.slice, ['a'], 1));
-
-slice(['a']);
-
-// $ExpectType (...args: string[]) => boolean
-expectType<(...args: string[]) => boolean>(bind.call(Boolean, null, String(), '2', '3', '4', '5'));
-
-// $ExpectType (...args: string[]) => boolean
-expectType<(...args: string[]) => boolean>(bind.apply(Boolean, [null, '1', '2', '3', '4', '5']));
```
Comment (ExE-Boss) [inline types/function-bind/ts3.2/function-bind-tests.ts:30]: This gets correctly inferred in **TS 3.3+** without needing `expectType<…>`:
<https://github.com/DefinitelyTyped/DefinitelyTyped/blob/b21787307a30f1d7b7d8e80bc58517e3576266bf/types/function-bind/ts3.9/function-bind-tests.ts#L26-L30>

**Review by ExE-Boss — APPROVED** [review]
FWIW, this is the diff of `@types/function‑bind` between **TS 3.2** and **TS 3.3+**:

File: types/w3c-screen-orientation/ts3.2/tsconfig.json [hunk 4]
```diff
@@ -1,24 +0,0 @@
-{
-    "compilerOptions": {
-        "module": "commonjs",
-        "lib": [
-            "es6",
-            "dom"
-        ],
-        "noImplicitAny": true,
-        "noImplicitThis": true,
-        "strictNullChecks": false,
```
Comment (ExE-Boss) [inline types/w3c-screen-orientation/ts3.2/tsconfig.json:10]: The only difference between **TS 3.2** and **TS 3.3+** for `@types/w3c‑screen‑orientation` is whether `strictNullChecks` is enabled.

**Review by ljharb — APPROVED** [review]

**Pull request merged**